src/services/goToDefinition.ts

```diff
@@ -99,7 +99,7 @@ namespace ts.GoToDefinition {
      */
     function symbolMatchesSignature(s: Symbol, calledDeclaration: SignatureDeclaration) {
         return s === calledDeclaration.symbol || s === calledDeclaration.symbol.parent ||
-            isVariableDeclaration(calledDeclaration.parent) && s === calledDeclaration.parent.symbol;
+            (isVariableDeclaration(calledDeclaration.parent) || isPropertyAssignment(calledDeclaration.parent)) && s === calledDeclaration.parent.symbol;

```

Comment:
would not it be better to check if the parent is callExpression or newExpression?